content/api/commands/origin.md

```diff
@@ -0,0 +1,467 @@
+callback will be executed. The test-runner will inject the Cypress runtime into
+this origin, and then send it code to evaluate in that origin, without violating
+the browser's same-origin policy.
+
+Secondly, it temporarily overrides the `baseUrl` configured in your
+[global configuration](/guides/references/configuration#Global) whilst inside
+the callback. So `cy.visit()` will navigate relative to this URL, not the
+configured `baseUrl`.
+
+**<Icon name="angle-right"></Icon> options** **_(Object)_**
+
+Pass in an options object to control the behavior of `cy.origin()`.
+
+| option | description                                                                                                                                                                                                    |
+| ------ | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| args   | Plain JavaScript object which will be serialized and sent from the primary origin to the secondary origin, where it will be deserialized and passed into the callback function as its first and only argument. |
+
+<Alert type="warning">
+
+The `args` object is the **only** mechanism via which data may be injected into
+the callback, the callback is **not** a closure and does not retain access to
+the JavaScript context in which it was declared.
+
+</Alert>
+
+**<Icon name="angle-right"></Icon> callbackFn** **_(Function)_**
+
+The function containing the commands to be executed in the secondary origin.
+
+This function will be stringified, sent to the Cypress instance in the secondary
+origin and evaluated. If the `args` option is specified, the deserialized args
+object will be passed into the function as its first and only argument.
+
+There are a number of limitations placed on commands run inside the callback,
+please see [Callback restrictions](#Callback-restrictions) section below for a
+full list.
+
+The function's return value is ignored. ???
+
+### Yields [<Icon name="question-circle"/>](/guides/core-concepts/introduction-to-cypress#Subject-Management)
+
+- `cy.origin()` yields `null`.
+- `cy.origin()` cannot be chained further.
+
+## Examples
+
+### Using dynamic data in a secondary origin
+
+Callbacks are executed inside an entirely separate instance of Cypress, so
+arguments must be transmitted to the other instance by means of serialization &
+deserialization. The interface for this mechanism is the `args` option.
+
+```js
+const sentArgs = { password: 'P@55w0rd!!!1!?!111one' }
+cy.origin(
+  'supersecurelogons.com',
+  // Send the args here...
+  { args: sentArgs },
+  // ...and receive them at the other end here!
+  (receivedArgs) => {
+    const { password } = receivedArgs
+    cy.visit('/login')
+    cy.get('input#password').type(password)
+    cy.contains('button', 'Login').click()
+  }
+)
```

Comment:
Seems more normal to send username and password. I'd also destructure `receivedArgs` earlier, but i assume you wrote it like you did to drive home the point that the args are received as params. so either way is probably fine.

```suggestion
const sentArgs = { username: 'username', password: 'P@55w0rd!!!1!?!111one' }
cy.origin(
  'supersecurelogons.com',
  // Send the args here...
  { args: sentArgs },
  // ...and receive them at the other end here!
  ({ username, password }) => {
    cy.visit('/login')
    cy.get('input#username').type(username)
    cy.get('input#password').type(password)
    cy.contains('button', 'Login').click()
  }
)
```

Reply:
Exactly correct!